modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java

```diff
@@ -0,0 +1,94 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST API methods for workflow.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_ADMIN, ROLE_USER})
+public class WorkFlowApi {
+
+    @Resource
+    private WorkflowProcessUtils workflowProcessUtils;
+
+    @Resource
+    private IdentityProviderManager identityProviderManager;
+
+    /**
+     * Query process status.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query process status by processInstanceId")
+    @GetMapping(value = "/workflow/state/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response queryWorkFlowById(@PathVariable("id") String processInstanceId) {
+        String workFlowState = workflowProcessUtils.getWorkFlowState(processInstanceId);
+        return Response.successResponse(List.of(workFlowState));
+    }
+
+    /**
+     * Query the tasks that need to be done by me.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query the tasks that need to be done by me")
+    @GetMapping(value = "/workflow/task/todo", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public List<WorkFlowTask> queryTodoTasks() {
+        String userId = identityProviderManager.getCurrentUserInfo().getUserId();
+        return workflowProcessUtils.todoTasks(userId);
+    }
+
+    /**
+     * Query the tasks I have completed.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query the tasks I have completed")
+    @GetMapping(value = "/workflow/task/done", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public List<WorkFlowTask> queryDoneTasks() {
+        String userId = identityProviderManager.getCurrentUserInfo().getUserId();
+        return workflowProcessUtils.doneTasks(userId);
+    }
+
+    /**
+     * Complete tasks by task ID.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Complete tasks by task ID.")
+    @PutMapping(value = "/workflow/task/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public void completeTask(@PathVariable("id") String taskId) {
```

Comment:
we need here to allow user to either retry task or just close the BPMN. 
We can call the API to have "retryWorkflow" and one more method "closeWorkflow". 
The "closeWorkflow" method will simply close the BPMN.
and "retryworkflow" method should work like how you have already implemented.